feat: Add Git Hook for Ruff (#52)
compsci-adl/courses-api: README.md

We welcome contributions to enhance Courses API! If you find any issues, have suggestions, or want to request a feature, please follow our [Contributing Guidelines](https://github.com/compsci-adl/.github/blob/main/CONTRIBUTING.md).

> This repository is configured with a pre-commit linting and formatting tool

Reviewer: Reword to better highlight what the pre-commit hook does.
```suggestion
This repository is configured with a pre-commit hook for automated linting and formatting.
```

To perform a commit from local, pre-commit must first be installed:
```
pip3 install pre-commit
```
OR if performing the action in bash:
```
python3 -m pip install pre-commit

> ```

Reviewer: We want to have pre-commit installed at a project level, so you can just do `uv add pre-commit --dev` and pre-commit should be installed when you do `uv sync`. You still need to do `pre-commit install` though to install the hook.

> Then, as recommended by [Official Pre-Commit Documentation](https://pre-commit.com/):

Reviewer: Reword to better highlight what this command is doing.
```suggestion
Install the git hook scripts:
```

```
pre-commit install
```

> This should be run every time you clone a project

Reviewer: Not really relevant.
```suggestion
```
